### ISSUE TITLE
Formatting for grouped metrics.

### ISSUE THREAD

**Issue by TezRomacH** [PyTorchLightning/pytorch-lightning#10097]
### Motivation

With Lightning, I use Tensorboard and ClearML, which allow you to group metrics in one tab. The "/" is used for grouping. In the screenshots from tb and ClearML you can see how they are grouped now.
<img width="500" alt="Screenshot 2021-10-23 at 22 14 43" src="https://user-images.githubusercontent.com/7549587/138589550-348d51d4-8ca5-48f8-8673-dd2e1d84a8d3.png">
<img width="499" alt="Screenshot 2021-10-23 at 22 22 04" src="https://user-images.githubusercontent.com/7549587/138589554-53fb3778-7a2e-406d-b5f1-c793d03ed3e0.png">
<img width="1127" alt="Screenshot 2021-10-23 at 22 23 46" src="https://user-images.githubusercontent.com/7549587/138589556-e8668916-9991-4713-a4fc-ee95c376c9d6.png">

### Pitch

1. The first is that when you select `on_step=True` and `on_epoch=True`, a postfix is appended to the end and the metrics by epoch and step are grouped into one in ClearML. I would like to split this group into two: `loss_step/<train or valid>` and `loss_epoch/<train or valid>`. Could the *name* parameter in *self.log* be formatted? What do you think? This would be something like `name="loss_{interval}/valid"`.

2. The second problem also involves formatting. When you pass a grouped metric to filename in ModelCheckpoint, Lightning will think "/" means to create a folder :slightly_smiling_face:  See screenshot. Would it be possible to add an escaping for slash and replace it with a lower underscore? 
<img width="539" alt="Screenshot 2021-10-23 at 22 24 12" src="https://user-images.githubusercontent.com/7549587/138589604-29cc73e9-aeb4-42b6-bbc7-07860bbfdcf1.png">
<img width="279" alt="Screenshot 2021-10-23 at 22 24 42" src="https://user-images.githubusercontent.com/7549587/138589605-3234abc0-64a0-4287-b8b6-d28d108eb0ae.png">

### Additional context

Similar problem, but with W&B in Lightning-Flash https://github.com/PyTorchLightning/lightning-flash/issues/818

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>
